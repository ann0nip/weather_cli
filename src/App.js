import React from "react";
import { makeStyles } from "@material-ui/core/styles";
import Container from "@material-ui/core/Container";
import Typography from "@material-ui/core/Typography";
import Box from "@material-ui/core/Box";
import Panel from "./components/Panel";

const useStyles = makeStyles((theme) => ({
  title: {
    fontSize: "2em",
    margin: 15,
    color: "#363636",
    fontFamily: "nunito",
    textDecoration: "underline",
  },
}));

function App() {
  const classes = useStyles();
  const [panels, setPanels] = React.useState([1]);
  return (
    <Container maxWidth="lg">
      <Box>
        <Typography className={classes.title} component="h1" align="center">
          Weather App
        </Typography>
        {panels.map((p) => {
          return <Panel />;
        })}
      </Box>
    </Container>
  );
}

export default App;
