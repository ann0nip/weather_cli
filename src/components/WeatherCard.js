import React, { useContext } from "react";
import PropTypes from "prop-types";
import { makeStyles } from "@material-ui/core/styles";
import FormGroup from "@material-ui/core/FormGroup";
import TextField from "@material-ui/core/TextField";
import Paper from "@material-ui/core/Paper";

import { getCurrent, getForecast } from "../utils/api";
import { WeatherContext } from "../context/WeatherContext";
import ButtonComponent from "./ButtonComponent";
import WeatherDetails from "./WeatherDetails";

const useStyles = makeStyles((theme) => ({
  paper: {
    backgroundColor: "#7db4fc",
    borderRadius: 6,
    width: 380,
    padding: 15,
    marginBottom: 15,
    marginTop: 15,
  },
  searchNav: {
    display: "flex",
    padding: 10,
    borderRadius: 6,
  },
  submitBtn: {
    backgroundColor: "#363636",
    color: "#FFF",
    marginLeft: 10,
  },
  formGroup: {
    justifyContent: "center",
    flexDirection: "row",
    paddingTop: 15,
  },
}));
const componentsStyles = {
  submitBtn: {
    backgroundColor: "#363636",
    color: "#FFF",
    marginLeft: 10,
  },
};

// WeatherCard.propTypes = {
//   classes: PropTypes.object.isRequired,
// };

export default function WeatherCard(props) {
  const classes = useStyles();
  const index = props.ix;
  const { weather, setWeather } = useContext(WeatherContext);
  const [city, setCity] = React.useState();

  const fetchData = async (event) => {
    event.preventDefault();
    const current = await getCurrent(city);
    const forecast = await getForecast(city);
    let weatherCopy = [...weather];
    weatherCopy[index] = { current, forecast };
    setWeather(weatherCopy);
  };

  return (
    <Paper className={classes.paper}>
      <FormGroup className={classes.formGroup}>
        <Paper elevation={0} className={classes.searchNav}>
          <TextField
            id="city_name"
            label="City Name"
            variant="outlined"
            size="small"
            onChange={(e) => {
              setCity(e.target.value);
            }}
          />
          <ButtonComponent
            onClick={fetchData}
            style={componentsStyles.submitBtn}
            text="Search"
          />
        </Paper>
      </FormGroup>
      {Object.keys(weather[index]).length > 0 && (
        <WeatherDetails data={weather[index]} />
      )}
    </Paper>
  );
}
