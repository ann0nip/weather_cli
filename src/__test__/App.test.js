import React from "react";
import { shallow } from "enzyme";

import App from "../App";
import WeatherCard from "../components/WeatherCard";
import Typography from "@material-ui/core/Typography";
import ButtonComponent from "../components/ButtonComponent";
import renderer from "react-test-renderer";

describe("<App />", () => {
  it("renders <WeatherCard /> component", () => {
    const wrapper = shallow(<App />);
    expect(wrapper.find(WeatherCard)).toHaveLength(1);

    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders the title", () => {
    const wrapper = shallow(<App />);
    expect(wrapper.find(Typography)).toHaveLength(1);
    wrapper.find(Typography).text("Weather App");
  });

  it("renders <ButtonComponent /> component", () => {
    const wrapper = shallow(<App />);
    expect(wrapper.find(ButtonComponent)).toHaveLength(1);
    wrapper.find(ButtonComponent).text("add city");

    const tree = renderer.create(<ButtonComponent />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("click in 'add city' should add a new <WeatherCard /> ", () => {
    const wrapper = shallow(<App />);
    const addCity = wrapper.find(ButtonComponent);

    addCity.simulate("click");
    expect(wrapper.find(WeatherCard)).toHaveLength(2);
  });
});
