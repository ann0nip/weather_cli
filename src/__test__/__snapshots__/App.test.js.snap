// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders <ButtonComponent /> component 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-text"
  disabled={false}
  onBlur={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiButton-label"
  />
  <span
    className="MuiTouchRipple-root"
  />
</button>
`;

exports[`<App /> renders <WeatherCard /> component 1`] = `
<div
  className="MuiContainer-root MuiContainer-maxWidthLg"
>
  <div
    className="MuiBox-root MuiBox-root-3"
  >
    <h1
      className="MuiTypography-root makeStyles-title-1 MuiTypography-body1 MuiTypography-alignCenter"
    >
      Weather App
    </h1>
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 MuiGrid-item"
      style={
        Object {
          "justifyContent": "flex-end",
        }
      }
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSizeLarge MuiButton-sizeLarge"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Add City
        </span>
      </button>
    </div>
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
    >
      <div
        className="MuiGrid-root makeStyles-gridCards-2 MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12"
      >
        <div
          className="MuiPaper-root makeStyles-paper-4 MuiPaper-elevation1 MuiPaper-rounded"
        >
          <div
            className="MuiFormGroup-root makeStyles-formGroup-7"
          >
            <div
              className="MuiPaper-root makeStyles-searchNav-5 MuiPaper-elevation0 MuiPaper-rounded"
            >
              <div
                className="MuiFormControl-root MuiTextField-root"
              >
                <label
                  className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-outlined"
                  data-shrink={false}
                  htmlFor="city_name"
                  id="city_name-label"
                >
                  City Name
                </label>
                <div
                  className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
                  onClick={[Function]}
                >
                  <input
                    aria-invalid={false}
                    autoFocus={false}
                    className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
                    disabled={false}
                    id="city_name"
                    onAnimationStart={[Function]}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    required={false}
                    type="text"
                  />
                  <fieldset
                    aria-hidden={true}
                    className="PrivateNotchedOutline-root-8 MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      className="PrivateNotchedOutline-legendLabelled-10"
                    >
                      <span>
                        City Name
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
              <button
                className="MuiButtonBase-root MuiButton-root MuiButton-text"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                style={
                  Object {
                    "backgroundColor": "#363636",
                    "color": "#FFF",
                    "marginLeft": 10,
                  }
                }
                tabIndex={0}
                type="button"
              >
                <span
                  className="MuiButton-label"
                >
                  Search
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
